--- a/vendor/doctrine/dbal/src/Schema/AbstractSchemaManager.php
+++ b/vendor/doctrine/dbal/src/Schema/AbstractSchemaManager.php
@@ -1344,6 +1344,12 @@ abstract class AbstractSchemaManager

         $indexes = [];
         foreach ($result as $indexKey => $data) {
+            // Fix for https://github.com/doctrine/dbal/issues/5306
+            // adapted from fix by Junaid Farooq at https://github.com/doctrine/dbal/pull/6811
+            if (! isset($data['columns'][0])) {
+                continue;
+            }
+            //
             $indexes[$indexKey] = new Index(
                 $data['name'],
                 $data['columns'],
